Update Adafruit_Arcada.h

rollalt opened the pull request:
Use benefit of Arvada to be able to display .GIFs on Itsybitsy M4.
Future extension to use MakeCode Arcade scheme with buttons.

rollalt committed:
Update Adafruit_Arcada.h

Use benefit of Arvada to be able to display .GIFs on Itsybitsy M4.
Future extension to use MakeCode Arcade scheme with buttons.

rollalt commented:
Will this be reviewed?
Thanks.

ladyada commented:
hiya please add details on how its wired up, and do not bump PR's - it can take a while for us to get to the 1200 github repos we maintain!

rollalt committed:
Update Adafruit_Arcada.h

Now with correct pin configuration.

rollalt commented:
@ladyada you are right, I've added the wrong pin configuration.
Now I've updated with correct pins.
The idea is to use "MakeCode Arcade" configuration. But for now I've added only the screen.
Please let me know if it is correct.
Thank you.

ladyada commented:
hi you need to add the buttons too, and test them!

Tijgerd commented:
Following:
https://learn.adafruit.com/makecode-arcade-with-samd51-m4/itsybitsy-m4-express

I think you need to add:
"A" = SCL
"B" = D7 
"Up" = D9
"Right" = D10 
"Left" =  D11
"Down" = D12 
"Menu" =  SDA

this will result in:
  #define ARCADA_BUTTONPIN_START        ??SDA??
  #define ARCADA_BUTTONPIN_SELECT       ??SDA??
  #define ARCADA_BUTTONPIN_A            ??SCL??
  #define ARCADA_BUTTONPIN_B            7
  #define ARCADA_BUTTONPIN_UP          9
  #define ARCADA_BUTTONPIN_DOWN   12
  #define ARCADA_BUTTONPIN_LEFT        11
  #define ARCADA_BUTTONPIN_RIGHT     10
  //#define ARCADA_USE_SD_FS   //don't know if this can be done?
  #define ARCADA_USE_JSON
  #define ARCADA_ACCEL_TYPE       ARCADA_ACCEL_NONE 

But then you also need to add some lines of code to Adafruit_Arcada.c in the function that reads the buttons..
